instagram/src/App.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import SearchBarContainer from './components/Search/SearchBarContainer';
+import PostContainer from './components/Post/PostContainer';
+import CommentsContainer from './components/Comments/CommentsContainer';
+import './App.css';
+import dummyData from './dummy-data.js';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {};
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({ dummyData, searchInput: '' });
+    }, 1000);
+    // this.setState({ dummyData, searchInput: '' });
+  }
+
+  handleSearchInput = event => {
+    this.setState({ searchInput: event.target.value });
+  };
+
+  searchPosts = event => {
+    event.preventDefault();
+    if (this.state.searchInput) {
+      this.setState({
+        dummyData: this.state.dummyData.filter(post => {
```

Comment:
You could move your call to filter into a separate variable outside of your `setState` call for easier debugging. That way your call to setState can look something like below.
```javascript
this.setState({dummyData: filteredPosts})
```